sql/core/src/main/scala/org/apache/spark/sql/execution/subquery.scala

```diff
@@ -157,7 +161,8 @@ case class InSubqueryExec(
       child = child.canonicalized,
       plan = plan.canonicalized.asInstanceOf[BaseSubqueryExec],
       exprId = ExprId(0),
-      resultBroadcast = null)
+      resultBroadcast = null,
+      result = null)
```

Comment:
hm, IIRC when it copies, it won't copy `@transient private var result: Array[Any] = _` .. I think we won't have to move `result` into the constructor (?).

Reply:
Hm, I think `copy` also work on transient variable.

Reply:
I thought it only works for `val`s. let me double check.

Reply:
Ah, yeah. It only copies the params in the constructor (I was confused w/ https://github.com/databricks/scala-style-guide#case-classes-and-immutability):

```scala
scala> case class A() {
     |     @transient var a: String = _
     | }
 defined class A
 
scala> val a = A()
a: A = A()

scala> a.a
res1: String = null

scala> a.a = "b"
a.a: String = b

scala> a.copy().a
res2: String = null

scala> a.a
res3: String = b
```

In any event, we won't bother moving `result` to constructor I guess.

Reply:
I tried to move it out of constructor, but there was some errors about `result` is null at the moment of preparing result. There might be some where we do `copy` it. In the case, we will lose `result` value.

Reply:
I see, okie. that's fine.